src/spikeinterface/core/generate.py

```diff
@@ -220,16 +220,236 @@ def add_synchrony_to_sorting(sorting, sync_event_ratio=0.3, seed=None):
             continue
         new_unit_indices[i] = rng.choice(units_not_used)
         units_used_for_spike[sample_index] = np.append(units_used_for_spike[sample_index], new_unit_indices[i])
+
     spikes_duplicated["unit_index"] = new_unit_indices
-    spikes_all = np.concatenate((spikes, spikes_duplicated))
-    sort_idxs = np.lexsort([spikes_all["sample_index"], spikes_all["segment_index"]])
-    spikes_all = spikes_all[sort_idxs]
+    sort_idxs = np.lexsort([spikes_duplicated["sample_index"], spikes_duplicated["segment_index"]])
+    spikes_duplicated = spikes_duplicated[sort_idxs]
 
-    sorting = NumpySorting(spikes=spikes_all, sampling_frequency=sorting.sampling_frequency, unit_ids=unit_ids)
+    synchronous_spikes = NumpySorting(spikes_duplicated, sorting.get_sampling_frequency(), unit_ids)
+    sorting = TransformSorting.add_from_sorting(sorting, synchronous_spikes)
 
     return sorting
 
 
+def generate_injected_sorting(
```

Comment:
Is Alessio not moving this already in https://github.com/SpikeInterface/spikeinterface/pull/2253?

Maybe we should merge that first to avoid this small conflict. I had requested over there to add the seed to the np.random there but we can do that later if there is no time. 

What do you think, @alejoe91 ?

Reply:
@yger I can drop my PR to avoid conflicts. Can you add some docstrics for this function?